app/services/evss_claim_service.rb

```diff
@@ -47,23 +43,19 @@ def upload_document(evss_claim_document)
     uploader.store!(evss_claim_document.file_obj)
     # the uploader sanitizes the filename before storing, so set our doc to match
     evss_claim_document.file_name = uploader.filename
-    EVSS::DocumentUpload.perform_async(auth_headers, @user.uuid, evss_claim_document.to_serializable_hash)
+    EVSS::DocumentUpload.perform_async(@user.uuid, evss_claim_document.to_serializable_hash)
   end
 
   def rating_info
-    client = EVSS::CommonService.new(auth_headers)
+    client = EVSS::EVSSCommon::Service.new(@user)
```

Comment:
Because of the EVSS headers are unique per user we have the issue of do we create an instance of a service for every user, as you've done here, or do we pass the user along in every service method call as unicorns do for the other EVSS services. `client.find_rating_info` needs a user's participant_id anyway so you could memoize one service instance and pass the user in the method as we do for EVSS letters https://github.com/department-of-veterans-affairs/vets-api/blob/f7e595291ed873baecc384728eaf594f8b44b5e1/app/controllers/v0/letters_controller.rb#L31. Although it probably doesn't make a huge difference that service instances are being created and garbage collected so often (this is how claims is working now).

When you can memoize it it's probably better to create a service that way, of course in a job I think you're creating a new instance all the time anyway so why not pass in the user on instantiation and have a cleaner api? Could be over thinking it 😄  @saneshark and @rlbaker might have ideas?

Reply:
@kreek I don't have any objections to memoizing the service instance. @lihanli Could `find_rating_info` use the participant_id from the user that was passed in on initialization? I think that would be worthy of refactoring if so.

Reply:
i can refactor `find_rating_info`, thanks for catching that.

```ruby
Benchmark.ms do
  5000.times { EVSS::Claims::Service.new(current_user) }
end
=> 2.4160190000657167
```
initializing a service seems to be pretty fast and since every request requires user headers i figured it would make the code cleaner by passing it in on init instead of adding the user to every method call